### PR TITLE
Alerting: Add frames for all queries and expressions

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**What this PR does / why we need it**:

This pull request is one of two pull requests to make the data frames for all queries and expressions in an alert rule available to the state package for rendering a graph. It renames Result to Cond, and creates an additional field called Results that is a map of Ref ID to data.Frames.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/ngalert/eval/eval.go [hunk 1]
```diff
@@ -73,12 +73,16 @@ func (e *invalidEvalResultFormatError) Unwrap() error {
 // ExecutionResults contains the unevaluated results from executing
 // a condition.
 type ExecutionResults struct {
-	Error error
+	// Cond contains the results of the condition
+	Cond data.Frames
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/eval/eval.go:77]: Let's use the full name.

**Review by yuri-tceretian — COMMENTED** [review]
What do you think about keeping the condition RefID in the ExecutionResults struct instead of the data frames. The potential benefit is that we can include it in the template and let user refer to it in the templates

**Comment by yuri-tceretian** [comment]
> Interesting. Condition would be a string field that contains the Ref ID?

yes.

**Comment by grobinson-grafana** [comment]
> > Interesting. Condition would be a string field that contains the Ref ID?
> 
> yes.

I think we can do this, but it will require much further refactoring because of how metadata is added to `Condition`. I would like to defer that work to another PR, if that's OK?

**Comment by yuri-tceretian** [comment]
> I think we can do this, but it will require much further refactoring because of how metadata is added to Condition. I would like to defer that work to another PR, if that's OK?

Yes. I just provided reasons for usage of RefID instead of Frame as Condition

**Review by yuri-tceretian — APPROVED** [review]

**Pull request merged**